.circleci/config.yml

```diff
@@ -69,6 +70,57 @@ commands:
           name: print installed packages
           command: pip freeze --all
 
+  dependencies-windows:
+    # You must still manually update the Docker image if any
+    # binary (non-Python) dependencies change.
+    description: "Check out and update Python dependencies."
+    steps:
+      - checkout
+
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - v6-win-dependencies-{{ checksum "setup.py" }}
+
+      - run:
+          name: install python
+          command: |
+            $ErrorActionPreference = "Stop"  # exit immediately on any error
+            choco install --side-by-side -y python --version 3.8
+          shell: powershell.exe
```

Comment:
Is it necessary to specify this? https://circleci.com/docs/using-windows#specifying-a-shell-with-the-windows-executor suggests powershell is the default

Reply:
Since I have provided bash.exe as the default shell (required for codecov/upload), I need to specify powershell.exe explicitly for commands that require it.
The virtual environment can only be activated through powershell.exe. So all the commands that require virtual env are run in powershell.